services/docker/docker-size.service.js

```diff
@@ -48,18 +78,54 @@ module.exports = class DockerSize extends BaseJsonService {
     return { message: prettyBytes(size) }
   }
 
-  async fetch({ user, repo, tag }) {
+  async fetch({ user, repo, tag, page }) {
+    page = page ? `&page=${page}` : ''
     return this._requestJson({
-      schema: buildSchema,
+      schema: tag ? buildSchema : pagedSchema,
       url: `https://registry.hub.docker.com/v2/repositories/${getDockerHubUser(
         user
-      )}/${repo}/tags/${tag}`,
+      )}/${repo}/tags${
+        tag ? `/${tag}` : '?page_size=100&ordering=last_updated'
+      }${page}`,
       errorMessages: { 404: 'image or tag not found' },
     })
   }
 
-  async handle({ user, repo, tag = 'latest' }) {
-    const data = await this.fetch({ user, repo, tag })
-    return this.constructor.render({ size: data.full_size })
+  transform({ tag, sort, data }) {
+    if (!tag && sort === 'date') {
+      return { size: data.results[0].full_size }
+    } else if (!tag && sort === 'semver') {
+      const [matches, versions] = data.reduce(
```

Comment:
Could we use an object/dict for `m` here instead of an array so that we can do a simple key lookup later on? 
Something like:
```js
      const [matches, versions] = data.reduce(
        ([m, v], d) => {
          m[d.name] = d.full_size
          v.push(d.name)
          return [m, v]
        },
        [{}, []]
      )
      const version = latest(versions)
      return { size: matches[version] }
```

Reply:
Yep, adjusted.